src/bin/add/args.rs

```diff
@@ -85,24 +90,28 @@ impl Args {
         } else if !crate_name.is_url_or_path() {
             let dependency = Dependency::new(&self.arg_crate);
 
-            if let Some(ref version) = self.flag_vers {
-                semver::VersionReq::parse(version)
-                    .chain_err(|| "Invalid dependency version requirement")?;
-                dependency.set_version(version)
-            } else if let Some(ref repo) = self.flag_git {
-                dependency.set_git(repo)
-            } else if let Some(ref path) = self.flag_path {
-                dependency.set_path(path.to_str().unwrap())
-            } else {
-                let dep = get_latest_dependency(&self.arg_crate, self.flag_allow_prerelease)?;
-                let v = format!(
-                    "{prefix}{version}",
-                    prefix = self.get_upgrade_prefix().unwrap_or(""),
-                    // If version is unavailable `get_latest_dependency` must have
-                    // returned `Err(FetchVersionError::GetVersion)`
-                    version = dep.version().unwrap_or_else(|| unreachable!())
-                );
-                dep.set_version(&v)
+            match (&self.flag_vers, &self.flag_path) {
```

Comment:
We need to return an error `if self.flag_git.is_some() && (self.flag_vers.is_some() || self.flag_path.is_some())`. See [failing test](https://travis-ci.org/killercup/cargo-edit/jobs/389539004#L1080).